firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -522,13 +525,21 @@ static long parseTokenExpirationTimestamp(String expiresIn) {
         : Long.parseLong(expiresIn.substring(0, expiresIn.length() - 1));
   }
 
-  private static void logFisCommunicationError(HttpURLConnection conn) {
+  private static void logFisCommunicationError(HttpURLConnection conn, String firebaseOptions) {
     String logString = readErrorResponse(conn);
     if (!TextUtils.isEmpty(logString)) {
       Log.w(FIS_TAG, logString);
+      Log.w(FIS_TAG, firebaseOptions);
     }
   }
 
+  private String availableFirebaseOptions(
+      @Nullable String appId, @NonNull String apiKey, @NonNull String projectId) {
+    return String.format(
+        "Firebase options used while communicating with Firebase server APIs: %s, %s, %s",
+        TextUtils.isEmpty(appId) ? "" : appId, apiKey, projectId);
```

Comment:
I don't think that this `TextUtils.isEmpty(appId) ? "" : appId` is necessary.
Just add appId and `String.format` will replace it by an empty String.

Reply:
Unless otherwise specified, passing a null argument to any method or constructor in this class will cause a NullPointerException to be thrown.

https://developer.android.com/reference/java/util/Formatter

Its safe to format on our end correctly vs depending on underlying Formatter impl.

Reply:
https://screenshot.googleplex.com/JVHfUAGMKGt